generate_qt_mappings.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python
+
+"""
+This script generates the Qt mapping file according to given Qt include
+directory
+
+Example usage :
+
+   $ ./generate_qt_mappings.py /usr/include/x86_64-linux-gnu/qt5 qt5_11.imp
+
+"""
+
+from __future__ import print_function
+import argparse
+import glob
+import os
+import re
+import sys
+
+
+def main(qt_include_dir, output_file):
+    symbols_map = []
+    includes_map = []
+
+    headers = glob.glob(os.path.join(args.qt_include_dir, '**/*[!.h]'))
+    for header in headers:
+        if os.path.isdir(header):
+            continue
+
+        class_name = os.path.basename(header)
+        module_name = os.path.basename(os.path.dirname(header))
+
+        symbols_map += ['{{ symbol: [ "{}", "private", '.format(class_name)
+                        + '"<{}>", "public" ] }}'.format(class_name)]
+
+        includes = []
+        with open(header, 'r') as f:
+            content = f.read()
+            includes = re.findall(r'#include "(.*)\.h"', content)
+
+        for include in includes:
+            includes_map += [
+                '{{ include: [ "@[\\"<]({}/)?{}\\\\.h[\\">]", '.format(
+                    module_name, include)
+                + '"private", "<{}>", "public" ] }}'.format(class_name)]
+
+    with open(args.output_file, 'w') as f:
+        print("# This file has been generated by the script {},"
+              "do not edit this file!\n[\n\t{}\n]".format(
+                  os.path.basename(__file__),
+                  ",\n\t".join(symbols_map + includes_map)), file=f)
```

Comment:
I'd still recommend using the `json` module. Something like this:

```
import json

# when building the output
symbols.append({"symbol": [class_name, "private", "<%s>" % class_name, "public" ]})
includes_map.append({"include": ["@[\\"<]({}/)?{}\\\\.h[\\">]", '.format(module_name, include), "private", "<{}>".format(class_name), "public"]}  # this is probably wrong now, I copied/pasted/edited without checking.

# when dumping to file
with open(args.output_file, 'w') as f:
    print("Some stuff for the header, file=f)
    json.dump(symbols, f, indent=2, sort_keys=True)
    json.dump(include_map f, indent=2, sort_keys=True)
```

That _might_ produce excessive indentation and line wrapping, but it's worth trying out to reduce the risk of misformatting JSON

Reply:
I tried using json module to serialize mapping like you describe above but I face a major issue: mapping file are not really json file.

Quote from the [documentation](https://github.com/include-what-you-use/include-what-you-use/blob/master/docs/IWYUMappings.md) 
> Single-word strings can be left un-quoted

Actually it HAS to be unquoted. If I use quote around `symbol` and `include` keyword (meaning the way json module generates file), mappings are not taking into account.

Reply:
@troopa81 Wow; I hadn't noticed that before. We use LLVM's YAML module to deserialize (JSON is allegedly a subset of YAML), and it has this strange behavior wrt quoting. It didn't occur to me that anything had to be done to unquote. I'll have to look into that; could you open a bug?

Reply:
Done https://github.com/include-what-you-use/include-what-you-use/issues/697

Reply:
:+1: